feat(ffe-lists): Add new modifier ffe-check-list__item--cross

selbekk opened the pull request:
This commit adds a new modifier for ffe-check-list list items to show that the item is not included in
whatever the list is about.

Resolves #471 

Not sure about the class name, nor how this should be implemented for the React component. We don't have any `ListItem` components I could add a flag to - should we perhaps add that?

Commit:
refactor(ffe-lists): Introduce list item classes

BREAKING CHANGE: This commit requires all list items to have a
corresponding list item class. The element classes introduced are as
follows:

- `ffe-bullet-list__item`
- `ffe-numbered-list__item`
- `ffe-stylized-numbered-list__item`
- `ffe-check-list__item`

To migrate, add the correct element class to each `li` in your
application. It should be a pretty simple search-replace for most cases,
but YMMV.

This change furthers our move away from element styling, to a design
system completely based on BEM.

Also, this commit adds a `cross` modifyer class for the `ffe-check-list__item`.

Finally, this commit also introduces element classes for the dd and dt elements
for `ffe-description-list` and `ffe-description-list-multicol`.

There are two new classes:

- `ffe-description-list__term` for `<dt />` tags
- `ffe-description-list__description` for `<dd />` tags

To migrate, please apply the correct class to the correct element
for all `ffe-description-list` instances.

ivarni commented:
The react component could take children as either `<li>`s or  `<ListItem>`s or you could just remove support for raw `li` elements. It shouldn't be a very difficult migration for consumers.

ivarni commented on packages/ffe-lists/less/regular-lists.less:
```diff
@@ -138,6 +141,14 @@
+    }
+
     &--bg-sand {
         > li {
```
Maybe this should be changed from styling the raw element and have a `.ffe-check-list__item` class instead? Would also mean not having to have a modifier for a non-existent class below, pluss we moved away from styling on raw elements somewhere around `v4` or something like that?

kwltrs commented on packages/ffe-lists/less/regular-lists.less:
```diff
@@ -18,6 +18,7 @@
 //     <li>Massevis av ferdige, testede komponenter</li>
 //     <li>Likt design som resten av SpareBank 1</li>
 //     <li>Høyere utviklingshastighet</li>
+//     <li class="ffe-check-list__item--cross">Null bugs</li>
```
I miss the base class `ffe-check-list__item`. I know it doesn't do anything, but that's why it is called a modifier in BEM.

selbekk replied:
yeah I agree, I'll implement what @ivarni suggested above

Commit:
feat(ffe-lists-react): Add list item components

This commit adds list item components for the different types of
lists in this package. These include:

- `BulletListItem`
- `CheckListItem`
- `DescriptionListTerm`
- `DescriptionListDescription`
- `NumberedListItem`
- `StylizedNumberedListItem`

These components add the correct class(es) to the list item, to make
migrating from the last version of `ffe-lists` even easier.

The `CheckListItem` also accepts a `isCross` prop, which triggers the
newly added `ffe-check-list__item--cross` modifier class.

selbekk commented:
So this was the biggest scope creep I've done in a while. :slightly_frowning_face: 

Note that `ffe-lists-react` won't get a major bump, because it still works as intended with `ffe-lists@^5.0.0`  AND `6.0.0`. The big difference is that with `6.0.0`, you'd have to manually add the classes to the list items and definition term/descriptions.

ivarni commented:
Just wondering if those commits to the same package should be squashed?

selbekk commented:
The three first ones? Meh

Edit: I don't mind either way - but I don't see a huge downside to keeping them separated

ivarni commented:
So that won't lead to three releases? How will the generated CHANGELOG look?

My god won't someone think of the changelog

selbekk commented:
Relax, there's only one release :laughing: or well, 2, ffe-lists@major and ffe-lists-react@minor

The changelog will hopefully look pretty nice - with nice migration guides.

selbekk commented:
To move this forward, I went ahead and squashed the commits per @ivarni's request.

Please let me know if there are further changes required before this PR can be approved.
